ghost/stats-service/lib/referrers.js

```diff
@@ -51,6 +53,83 @@ class ReferrersStatsService {
 
         return [...map.values()].sort((a, b) => b.paid_conversions - a.paid_conversions);
     }
+
+    /**
+     * Return a list of all the attribution sources, with their signup and conversion counts on each date
+     * @returns {Promise<{data: AttributionCountStat[], meta: {}}>}
+     */
+    async getReferrersHistory() {
+        const paidConversionEntries = await this.fetchAllPaidConversionSources();
+        const signupEntries = await this.fetchAllSignupSources();
+
+        const allEntries = signupEntries.map((entry) => {
+            return {
+                ...entry,
+                paid_conversions: 0,
+                date: moment(entry.date).format('YYYY-MM-DD')
+            };
+        });
+
+        paidConversionEntries.forEach((entry) => {
+            const entryDate = moment(entry.date).format('YYYY-MM-DD');
+            const existingEntry = allEntries.find(e => e.source === entry.source && e.date === entryDate);
+
+            if (existingEntry) {
+                existingEntry.paid_conversions = entry.paid_conversions;
+            } else {
+                allEntries.push({
```

Comment:
We probably need to sort in the end because now the dates won't be sorted correctly any longer

Reply:
@SimonBackx Good catch! 🎣 Updated and also added test for sorted dates